module/modmanager/manager.go

```diff
@@ -633,7 +633,10 @@ func (mgr *Manager) ResolveImplicitDependenciesInConfig(ctx context.Context, con
 			// continue if this component is not being provided by a module.
 			continue
 		}
-		if !slices.ContainsFunc(conf.Modified.Modules, func(elem config.Module) bool {
+
+		//nolint:gocritic
+		deltaModules := append(conf.Modified.Modules, conf.Added.Modules...)
```

Comment:
Without this line, on reconfiguration after rename we run into a new bug where modular components that existed in the previous config never get (re-)added to the resource graph. These components get removed on module shutdown but they aren't diffed as new components (since they exist in both the old and new config) so don't get added back. Renaming modules also causes them to be `conf.Modules.Added`, not `conf.Modules.Modified`, so these components exist in a state where they are not new, but their module is. 

In`ResolveImplicitDependenciesInConfig` we must now also check for components that exist in this state (from an `Added` module but currently an unadded component) and add them back in.

Reply:
Kind of a hack, open to suggestions to fix this bug but the cleanest solution in my eyes would be to not consider a renamed module as new, but that requires refactors to how we diff

Reply:
I think this is probably the best we can do for now until we have better resource states internally (some ability to mark a resource as needing rebuilding due to its module being either modified or just added).

Reply:
can we add a ticket for that? `add ability to mark a resource as needing rebuilding due to its module being either modified or just added`